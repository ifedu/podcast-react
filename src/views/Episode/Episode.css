.Episode {
  display: flex;
  justify-content: space-between;
}

.Episode .info {
  box-shadow: 0 1px 2px 2px #ddd;
  padding: 16px;
  width: 800px;
}

.Episode .info .title {
  font-size: 20px;
  font-weight: bold;
  margin-bottom: 12px;
}

.Episode .info .description {
  border-bottom: 1px solid #ccc;
  font-style: italic;
  margin-bottom: 16px;
  padding-bottom: 16px;
}

.Episode audio {
  height: 32px;
  width: 100%;
}
