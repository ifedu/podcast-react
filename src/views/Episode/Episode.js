import React from 'react';
import { useLocation } from 'react-router-dom';

import { PodcastInfo } from '../../components/PodcastInfo/PodcastInfo';

import './Episode.css';

export function Episode() {
  const { state: { episode, podcastInfo, song } } = useLocation();

  return (
    <div className='Episode' data-testid='episode'>
      <PodcastInfo info={podcastInfo.results[0]} song={song}/>

      <div className='info'>
        <div className='title'>{episode.trackName}</div>
        <div className='description' dangerouslySetInnerHTML={{ __html: episode.description }}></div>
        <div className='audio'>
          <audio controls>
            <source src={episode.episodeUrl}/>
          </audio>
        </div>
      </div>
    </div>
  );
}
