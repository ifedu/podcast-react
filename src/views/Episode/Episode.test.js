import { cleanup, render, screen, waitFor } from '@testing-library/react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import renderer from 'react-test-renderer' ;

import { Episode } from './Episode';

const getContent = (pathname = '/') => {
  const state = {
    episode: {},
    podcastInfo: {
      resultCount: 1,
      results: [
        {
          collectionName: 'Podcast Name',
          description: 'Podcast Description',
        },
      ],
    },
    song: {
      summary: {},
    },
  };

  return <MemoryRouter initialEntries={[{ pathname, state }]}>
    <Routes>
      <Route path='/podcast/1/episode/1' element={
        <Episode/>
      }></Route>
    </Routes>
  </MemoryRouter>;
}

describe('Episode', () => {
  const view = getContent('/podcast/1/episode/1');

  afterEach(cleanup);

  it('renders correctly', () => {
    const tree = renderer.create(view).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the episode', async() => {
    render(view);

    await waitFor(() => {
      const podcast = screen.getByTestId('episode');
      expect(podcast).toBeInTheDocument();
    });
  });
});
