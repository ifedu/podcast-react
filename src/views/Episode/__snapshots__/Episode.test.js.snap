// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Episode renders correctly 1`] = `
<div
  className="Episode"
  data-testid="episode"
>
  <div
    className="PodcastInfo"
    data-testid="podcast-info"
  >
    <a
      href="/podcast/undefined"
      onClick={[Function]}
    >
      <div
        className="section picture"
      >
        <img
          alt="All Songs Considered"
        />
      </div>
      <div
        className="section title"
      >
        <span>
          Podcast Name
        </span>
        <div>
          by 
        </div>
      </div>
    </a>
    <div
      className="section description"
    >
      <span>
        Description
      </span>
      <div />
    </div>
  </div>
  <div
    className="info"
  >
    <div
      className="title"
    />
    <div
      className="description"
      dangerouslySetInnerHTML={
        Object {
          "__html": undefined,
        }
      }
    />
    <div
      className="audio"
    >
      <audio
        controls={true}
      >
        <source />
      </audio>
    </div>
  </div>
</div>
`;
